develop/blockchains/custom-blockchains/benchmarking.md

```diff
@@ -1,4 +1,183 @@
+
+```toml title="Cargo.toml"
+runtime-benchmarks = [
+    "frame-benchmarking/runtime-benchmarks",
+    "frame-support/runtime-benchmarks",
+    "frame-system/runtime-benchmarks",
+    "sp-runtime/runtime-benchmarks",
+]
+```
+
+Lastly, ensure that `frame-benchmarking` is included in `std = []`: 
+
+```toml title="Cargo.toml"
+std = [
+    # ...
+    "frame-benchmarking?/std",
+    # ...
+]
+```
+
+Once complete, you have the required dependencies for writing benchmark tests for your pallet.
+
+### Write Benchmark Tests
+
+Create a `benchmarking.rs` file in your pallet's `src/`. Your directory structure should look similar to the following:
+
+```
+my-pallet/
+├── src/
+│   ├── lib.rs          # Main pallet implementation
+│   └── benchmarking.rs # Benchmarking
+└── Cargo.toml
+```
+
+With the directory structure set, you can use the [`polkadot-sdk-parachain-template`](https://github.com/paritytech/polkadot-sdk-parachain-template/tree/master/pallets){target=\_blank} to get started as follows:
+
+```rust title="benchmarking.rs (starter template)"
+--8<-- 'https://raw.githubusercontent.com/paritytech/polkadot-sdk-parachain-template/refs/tags/v0.0.2/pallets/template/src/benchmarking.rs'
+```
+
+In your benchmarking tests, employ these best practices:
+
+- **Write custom testing functions** - the function `do_something` in the preceding example is a placeholder. Similar to writing unit tests, you must write custom functions to benchmark test your extrinsics. Access the mock runtime and use functions such as `whitelisted_caller()` to sign transactions and facilitate testing
+- **Use the `#[extrinsic_call]` macro** - this macro is used when calling the extrinsic itself and is a required part of a benchmark testing function. See the [`extrinsic_call Rust docs](https://paritytech.github.io/polkadot-sdk/master/frame_benchmarking/v2/index.html#extrinsic_call-and-block){target=_blank} for more details
+- **Validate extrinsic behavior** - the `assert_eq` expression ensures that the extrinsic is working properly within the benchmark context
+
+### Add Benchmarks to Runtime
+
+Before running the benchmarking tool, you must integrate benchmarks with your runtime as follows:
+
+1. Create a `benchmarks.rs` file. This file should contain the following macro, which registers all pallets for benchmarking, as well as their respective configurations: 
+    ```rust title="benchmarks.rs"
+    --8<-- 'code/develop/blockchains/custom-blockchains/benchmarking/frame-benchmark-macro.rs'
+    ```
+    For example, to register a pallet named `pallet_parachain_template` for benchmark testing, add it as follows: 
+    ```rust title="benchmarks.rs" hl_lines="3"
+    --8<-- 'code/develop/blockchains/custom-blockchains/benchmarking/frame-benchmark-macro.rs::3'
+    );
+    ```
+
+    ???warning "Updating `define_benchmarks!` macro is required"
+        If the pallet isn't included in the `define_benchmarks!` macro, the CLI cannot access and benchmark it later.
```

Comment:
Stuff like this can just be expanded automatically. It's not like it's saving space by making it expandable, so lets just present the info off the bat